google-cloud-storage/src/test/java/com/google/cloud/storage/BucketTest.java

```diff
@@ -649,6 +650,20 @@ public void testDeleteAcl() throws Exception {
     assertTrue(bucket.deleteAcl(User.ofAllAuthenticatedUsers()));
   }
 
+  @Test
+  public void testDeleteLifecycleRules() {
+    Map<LifecycleRule, Boolean> expectedResults = ImmutableMap.of(LIFECYCLE_RULES.get(0), true);
+    expect(storage.getOptions()).andReturn(mockOptions).times(1);
+    expect(storage.deleteLifecycleRules(FULL_BUCKET_INFO.getName(), LIFECYCLE_RULES.get(0)))
+        .andReturn(expectedResults);
+    replay(storage);
+    initializeBucket();
+    Map<LifecycleRule, Boolean> actualResults = bucket.deleteLifecycleRules(LIFECYCLE_RULES.get(0));
+    assertThat(actualResults).hasSize(1);
+    assertThat(actualResults).containsKey(LIFECYCLE_RULES.get(0));
+    assertThat(actualResults).containsEntry(LIFECYCLE_RULES.get(0), true);
+  }
+
```

Comment:
your test does not cover case of deleting non existing rule

Reply:
done already added